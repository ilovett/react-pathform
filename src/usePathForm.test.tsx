import React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { PathFormProvider, usePathForm } from '.';

function wrapper({ children }: { children: React.ReactChildren }) {
  return (
    <PathFormProvider
      initialRenderValues={{
        nested: {
          items: [{ name: 'A' }, { name: 'B' }, { name: 'C' }, { name: 'D' }],
        },
      }}
    >
      {children}
    </PathFormProvider>
  );
}

describe('array', () => {
  function render() {
    return renderHook(() => usePathForm(), { wrapper });
  }

  describe('append', () => {
    it('adds items to the end of an array', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.append(['nested', 'items'], { name: 'LAST' });

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });

  describe('insert', () => {
    it('inserts items into an array at a given index', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.insert(['nested', 'items'], 1, { name: 'AA' }, { name: 'AB' });

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });

  describe('move', () => {
    it('moves items in an array from an index to another index', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.move(['nested', 'items'], 0, 4);

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });

  describe('prepend', () => {
    it('prepends an item to the beginning of an array', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.prepend(['nested', 'items'], { name: 'FIRST' });

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });

  describe('splice', () => {
    it('splices as expected', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.splice(['nested', 'items'], 1, 2, { name: 'BB' }, { name: 'CC' });

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });

  describe('remove', () => {
    it('removes items from an array as expected', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.remove(['nested', 'items'], 1, 2);

      expect(result.current.getValues()).toMatchSnapshot();
    });

    it('removes one item from an array by default', () => {
      const { result } = render();

      expect(result.current.getValues()).toMatchSnapshot();

      result.current.array.remove(['nested', 'items'], 1);

      expect(result.current.getValues()).toMatchSnapshot();
    });
  });
});
