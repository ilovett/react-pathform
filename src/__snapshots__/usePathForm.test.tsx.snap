// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array append adds items to the end of an array 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array append adds items to the end of an array 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
      Object {
        "name": "LAST",
      },
    ],
  },
}
`;

exports[`array insert inserts items into an array at a given index 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array insert inserts items into an array at a given index 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "AA",
      },
      Object {
        "name": "AB",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array move moves items in an array from an index to another index 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array move moves items in an array from an index to another index 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
      Object {
        "name": "A",
      },
    ],
  },
}
`;

exports[`array prepend prepends an item to the beginning of an array 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array prepend prepends an item to the beginning of an array 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "FIRST",
      },
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array remove removes items from an array as expected 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array remove removes items from an array as expected 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array remove removes one item from an array by default 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array remove removes one item from an array by default 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array splice splices as expected 1`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "B",
      },
      Object {
        "name": "C",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;

exports[`array splice splices as expected 2`] = `
Object {
  "nested": Object {
    "items": Array [
      Object {
        "name": "A",
      },
      Object {
        "name": "BB",
      },
      Object {
        "name": "CC",
      },
      Object {
        "name": "D",
      },
    ],
  },
}
`;
